app/uk/gov/hmrc/apidocumentation/controllers/LoggedInUserProvider.scala

```diff
@@ -39,11 +38,8 @@ class LoggedInUserProvider @Inject()(config: ApplicationConfig,
   def resolveUser(id: String)(implicit ctx: ExecutionContext, hc: HeaderCarrier): Future[Option[Developer]] =
     sessionService.fetch(id).map(_.map(_.developer))
 
-  def fetchLoggedInUser()(implicit request: Request[_], hc: HeaderCarrier): Future[Option[Developer]] = {
-
-    val oToken = tokenAccessor.extract(request)
-
-    oToken match {
+  def fetchLoggedInUser()(implicit request: Request[_], hc: HeaderCarrier): Future[Option[Developer]] = 
```

Comment:
Looks like too much pattern matching.  Can we manage either a for comp or simple map/flatmap ?  Would need to see function return types to be sure.

Reply:
That code hasn't changed, except to remove the redundant val. I'm minded to leave it as it is so as to avoid introducing bugs that aren't not related to the upgrade of Play.